katalog/elasticsearch-triple/triple.yml

```diff
@@ -9,5 +13,66 @@ spec:
       containers:
         - name: elasticsearch
           env:
-            - name: discovery.zen.ping.unicast.hosts
-              value: "elasticsearch-0.elasticsearch,elasticsearch-1.elasticsearch,elasticsearch-2.elasticsearch"
+            - name: cluster.initial_master_nodes
+              value: "elasticsearch-0,elasticsearch-1,elasticsearch-2"
+          readinessProbe:
+            $patch: replace
+            failureThreshold: 3
+            initialDelaySeconds: 45
+            periodSeconds: 10
+            successThreshold: 3
+            timeoutSeconds: 5
+            exec:
+              command:
+                - sh
+                - -c
+                - |
+                  #!/usr/bin/env bash -e
+                  # If the node is starting up wait for the cluster to be ready (request params: "wait_for_status=green&timeout=1s" )
+                  # Once it has started only check that the node itself is responding
+                  START_FILE=/tmp/.es_start_file
+
+                  http () {
+                    local path="${1}"
+                    local args="${2}"
+                    set -- -XGET -s
+
+                    if [ "$args" != "" ]; then
+                      set -- "$@" $args
+                    fi
+
+                    if [ -n "${ELASTIC_USERNAME}" ] && [ -n "${ELASTIC_PASSWORD}" ]; then
+                      set -- "$@" -u "${ELASTIC_USERNAME}:${ELASTIC_PASSWORD}"
+                    fi
+
+                    curl --output /dev/null -k "$@" "http://127.0.0.1:9200${path}"
+                  }
+
+                  if [ -f "${START_FILE}" ]; then
+                    echo 'Elasticsearch is already running, lets check the node is healthy'
+                    HTTP_CODE=$(http "/" "-w %{http_code}")
+                    RC=$?
+                    if [[ ${RC} -ne 0 ]]; then
+                      echo "curl --output /dev/null -k -XGET -s -w '%{http_code}' \${BASIC_AUTH} http://127.0.0.1:9200/ failed with RC ${RC}"
+                      exit ${RC}
+                    fi
+                    # ready if HTTP code 200, 503 is tolerable if ES version is 6.x
+                    if [[ ${HTTP_CODE} == "200" ]]; then
+                      exit 0
+                    elif [[ ${HTTP_CODE} == "503" && "7" == "6" ]]; then
+                      exit 0
```

Comment:
Sorry for my OCD, we can safely delete these lines as they are never used:
```
elif [[ ${HTTP_CODE} == "503" && "7" == "6" ]]; then
  exit 0
```

Reply:
Nice catch up!